### ISSUE TITLE
Constructor signature - new Pusher(...)

### ISSUE THREAD

**Issue by leggetter** [pusher/pusher-http-php#66]
We want to keep the constructor as simple as possible. I'd suggest the use of the current `Config` object adds additional and unnecessary complexity.

A key part of doing this will be providing default values for things that were otherwise extracted from the URL:

* Base URL
* Encrypted (HTTP or HTTPS) - **not actually part of the current Config, but probably should be**

My vote is for one of the following:

---

## 1. Directly passing required parameters as arguments

```php
$pusher = new Pusher('my-app-id', 'my-app-key', 'my-app-secret');
```

Additional options are added by a 4th `options` parameter:

```php
$pusher = new Pusher('my-app-id', 'my-app-key', 'my-app-secret', array('encrypted' => true));
```

## 2. Using an Associative Array with named parameters

```php
$pusher = new Pusher(array(appId => 'id_value', appKey' => 'key_value', 'appSecret' => 'secret_value', ...);
```

Optional parameters are just added to the array:

```php
$pusher = new Pusher(array(appId => 'id_value', appKey' => 'key_value', 'appSecret' => 'secret_value', 'encrypted' => true);
```

The benefit of this approach is that it reduces the changes of getting key and secret mixed up.

---

What do others think? @zimbatm

**Comment by tanuck** [comment]
I would go with the first option. As long as any boolean type parameters are kept in the 4th argument array. I always feel passing boolean parameters in that way is much better for readability. I do think the `Pusher\Config` class should be kept though!

**Comment by leggetter** [comment]
@tanuck To clarify: would you'd prefer:

```php
use Pusher\Pusher;
use Pusher\Config;

$pusher = new Pusher(new Config('my-app-id', 'my-app-key', 'my-app-secret', array('encrypted' => true)));
```

For me:

* the two `use` statements is one too many and an unnecessary additional step
* it results in a bunch of extra parenthesis that people can make mistakes with results in one more annoyance on the way to getting things working

But happy to add this as a +1 for the `Config` based constructor.

**Comment by tanuck** [comment]
No. That option is a bit too cluttered for me. And I agree about the 2 use statements. I was happy with your first example from option 1:

```php
$pusher = new Pusher('my-app-id', 'my-app-key', 'my-app-secret', array('encrypted' => true));
```

But keep the Config class for internal use. Instead of cluttered the Pusher class with additional config related properties.

It would be good to decide on this and merge the implementation before looking at #68

**Comment by leggetter** [comment]
:+1: Yep - good call on keeping the internal class.

Agree RE #68 too.

**Comment by tanuck** [comment]
Okay :+1:

**Comment by leggetter** [comment]
Fixed in #69.
